src/main/java/org/gradle/profiler/mutations/support/ProjectCombinations.java

```diff
@@ -0,0 +1,25 @@
+package org.gradle.profiler.mutations.support;
+
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+
+public class ProjectCombinations {
```

Comment:
Why does this belong in a package called `support`? This is part of the mutators.

If we could merge the two project-dependency-change-related mutators, we could move this to a nested subclass, and that'd look a lot better I think.

Reply:
Mutations package is really huge, and it's hard to find classes there. I created this package so these support classes, that are not mutators, are easy to find. They are still under mutations package.

I will check how easy it is to merge two classes into one.

Reply:
I merged mutator classes into one. Code wise it is probably easier to understand. I also moved ProjectCombinations as a subclass of now one mutator class. 

But I kept the support class, that way since it's a bit complex logic and it's much nicer to test that if it is separate. And since it's easier to test that when separate unit, that is a good flag for me that it might deserve a separate class. WDYT?

Reply:
I like that `ProjectCombinations` is now a nested class. I think that makes this support class even more of an orphan that should be reunited with its folks.

Reply:
removed support class